server/enterprise/message_export/global_relay_export/global_relay_export.go

```diff
@@ -70,12 +72,28 @@ type Message struct {
 	PreviewsPost   string
 }
 
-func GlobalRelayExport(rctx request.CTX, posts []*model.MessageExport, db shared.MessageExportStore, fileAttachmentBackend filestore.FileBackend, dest io.Writer, templates *templates.Container) ([]string, int, error) {
-	var warningCount int
+type Params struct {
+	ExportType            string
+	Posts                 []*model.MessageExport
+	BatchPath             string
+	Config                *model.Config
+	Db                    shared.MessageExportStore
+	FileAttachmentBackend filestore.FileBackend
+	ExportBackend         filestore.FileBackend
+	Templates             *templates.Container
+}
+
+func GlobalRelayExport(rctx request.CTX, p Params) (results shared.RunExportResults, err error) {
+	tmpFile, err := os.CreateTemp("", "")
+	if err != nil {
+		return results, fmt.Errorf("unable to open the temporary export file: %w", err)
```

Comment:
Do we expect to return partial results in case of error? Otherwise, I wonder if it would be less ambiguous to return the empty value.

Reply:
@streamer45 I've been following the rule of "if the err is not nil, the returned value isn't valid" so sending back whatever we have is fine. But I can see the point of returning an empty value too for clarity.
Is returning empty the rule, or should it be? I'm fine either way.

Reply:
I don't think there are any rules, just opinions and preferences at this point. No action is needed here.